src/main/java/ch/uzh/ifi/access/course/model/Course.java

```diff
@@ -34,7 +34,9 @@ public Course() {
     public void set(Course other) {
         //this.directory = other.directory;
         //this.gitURL = other.gitURL;
-        this.gitHash = other.gitHash;
+        if (other.gitHash != null) {
```

Comment:
Oh ups! Ja der `gitHash` muss im `update()` gesetzt werden und nicht im `set()`. Genau gleich wie ich das bei dem `Exercise` auch gemacht habe. May bad! Ich habe nur das vom `Exercise` programmiert und dann am schluss gedacht why the heck not und dem `Course` auch ein `gitHash` gegeben aber einfach am falschen ort :P

Reply:
Aha, alles klar :P